include/engine/routing_algorithms/routing_base.hpp

```diff
@@ -90,15 +90,8 @@ template <class DataFacadeT, class Derived> class BasicRoutingInterface
             if (new_distance < upper_bound)
             {
                 // if loops are forced, they are so at the source
-                if (new_distance >= 0 &&
-                    (!force_loop_forward || forward_heap.GetData(node).parent != node) &&
-                    (!force_loop_reverse || reverse_heap.GetData(node).parent != node))
-                {
-                    middle_node_id = node;
-                    upper_bound = new_distance;
-                }
-                else
-                {
+                if ((force_loop_forward && forward_heap.GetData(node).parent == node) ||
```

Comment:
The switch of the instructions here is not entirely the same.

Shouldn't the first check include a `new_distance < 0 ||`? From what I've seen you inverted the check and switched the if/else branches? But the new conditions are not entirely the same. On `< 0` distances, the now `if` branch should be executed, not the `else` part. Or not?